contracts/TalentLayerEscrow.sol

```diff
@@ -383,16 +503,155 @@ contract TalentLayerEscrow is Ownable {
         Transaction storage transaction = transactions[_transactionId];
 
         require(transaction.receiver == msg.sender, "Access denied.");
+        require(transaction.status == Status.NoDispute, "The transaction shouldn't be disputed.");
         require(transaction.amount >= _amount, "Insufficient funds.");
 
         transaction.amount -= _amount;
         _reimburse(transaction, _amount);
 
-        emit Payment(PaymentType.Reimburse, _amount, transaction.token, transaction.serviceId);
+        emit Payment(_transactionId, PaymentType.Reimburse, _amount, transaction.token, transaction.serviceId);
 
         _distributeMessage(transaction.serviceId, transaction.amount);
     }
 
+    /** @notice Allows the sender of the transaction to pay the arbitration fee to raise a dispute.
+     *  Note that the arbitrator can have createDispute throw, which will make this function throw and therefore lead to a party being timed-out.
+     *  This is not a vulnerability as the arbitrator can rule in favor of one party anyway.
+     *  @param _transactionId Id of the transaction.
+     */
+    function payArbitrationFeeBySender(uint256 _transactionId) public payable {
+        Transaction storage transaction = transactions[_transactionId];
+        uint256 arbitrationCost = transaction.arbitrator.arbitrationCost(transaction.arbitratorExtraData);
+
+        require(
+            transaction.status < Status.DisputeCreated,
+            "Dispute has already been created or because the transaction has been executed."
+        );
+        require(msg.sender == transaction.sender, "The caller must be the sender.");
+
+        transaction.senderFee += msg.value;
+        // The total fees paid by the sender should be at least the arbitration cost.
+        require(transaction.senderFee >= arbitrationCost, "The sender fee must cover arbitration costs.");
+
+        transaction.lastInteraction = block.timestamp;
+
+        emit ArbitrationFeePayment(_transactionId, ArbitrationFeePaymentType.Pay, Party.Sender, msg.value);
+
+        // The receiver still has to pay. This can also happen if he has paid, but arbitrationCost has increased.
```

Comment:
This can actually happen. Only arbitrator address and extraData are linked to the transaction but the arbitration cost is not, so it could have changed since the first party paid. 
It would be a bit tricky to link also arbitration cost to a transaction, because the arbitrator has to check that the right amount is passed when creating a dispute, but it has no info on transactions. Do you think this would make sense?